oratorio/static/scripts.js

```diff
@@ -78,9 +78,14 @@ function upload(blob){
         var id_token = getCookie("id_token")
 
         xhr.onload = function () {
-            console.log('DONE', xhr.readyState); // readyState will be 4
-            rec_id = xhr.response;
-            window.location = "result?rid=" + rec_id;
+            if (xhr.status == 200) {
+              rec_id = parseInt(xhr.response);
+              window.location = "result?rid=" + rec_id;
+            } else {
+              // An error occurred, return to index for now
+              // TODO: display error in some way to the user
+              window.location = "";
```

Comment:
the way to do this would be pass it get parameters as part of the location and have the onload parse those into an error message. Although that is prone to potential scripting errors, but its something to be fixed later.